src/setupWorker/glossary.ts

```diff
@@ -118,10 +118,10 @@ export interface SetupWorkerInternalContext {
      * Adds an event listener on the given target.
      * Returns a clean-up function that removes that listener.
      */
-    addListener<E extends Event>(
+    addListener(
       target: EventTarget,
       eventType: string,
-      listener: (event: E) => void,
+      listener: ((event: MessageEvent) => void) | EventListener,
```

Comment:
The `listener` doesn't have to accept `MessageEvent` only. This is a general shorthand function to create/remove event listeners on any target:

```js
context.events.addListener(window, 'beforeunload')
```

I'd expect this call signature to throw type violations.